tests/src/testnonlineareas.cpp

```diff
@@ -0,0 +1,161 @@
+  using namespace Dune::Functions::BasisFactory;
+  auto basis = Ikarus::makeBasis(gridView, power<2>(lagrange<1>(), FlatInterleaved()));
+
+  auto sk      = skills(nonLinearElastic(reducedMat), eas(4));
+  using FEType = decltype(makeFE(basis, sk));
+  std::vector<FEType> fes;
+
+  for (auto&& ge : elements(gridView)) {
+    fes.emplace_back(makeFE(basis, sk));
+    fes.back().bind(ge);
+  }
+
+  auto basisP = std::make_shared<const decltype(basis)>(basis);
+  Ikarus::DirichletValues dirichletValues(basisP->flat());
+
+  // fix left edge (x=0)
+  dirichletValues.fixBoundaryDOFs([&](auto& dirichletFlags, auto&& localIndex, auto&& localView, auto&& intersection) {
+    if (std::abs(intersection.geometry().center()[0]) < 1e-8)
+      dirichletFlags[localView.index(localIndex)] = true;
+  });
+
+  using SparseAssembler = SparseFlatAssembler<decltype(fes), decltype(dirichletValues)>;
+  SparseAssembler sparseFlatAssembler(fes, dirichletValues);
+  auto sparseAssemblerAM = makeAssemblerManipulator(sparseFlatAssembler);
+
+  Eigen::VectorXd d;
+  d.setZero(basis.flat().size());
+  double lambda = 0.0;
+
+  auto req = typename FEType::Requirement(d, lambda);
+
+  sparseAssemblerAM->bind(req, Ikarus::AffordanceCollections::elastoStatics, DBCOption::Full);
+
+  // Apply constant point load at the top right corner
+  Dune::FieldVector<double, gridDim> topRightPos{L, h};
+  const auto globalIndices = utils::globalIndexFromGlobalPosition(basis.flat(), topRightPos);
+  auto pointLoad           = [&](const auto&, const auto& par, auto, auto, Eigen::VectorXd& vec) -> void {
+    auto loadFactor = par.parameter();
+    vec[globalIndices[1]] -= -loadFactor * 1.0;
+  };
+  sparseAssemblerAM->bind(pointLoad);
+
+  auto linSolver = LinearSolver(SolverTypeTag::sd_UmfPackLU);
+
+  AffordanceCollection elastoStaticsNoScalar(VectorAffordance::forces, MatrixAffordance::stiffness);
+
+  auto nonOp = NonLinearOperatorFactory::op(sparseAssemblerAM, elastoStaticsNoScalar, sparseAssemblerAM->dBCOption());
+
+  constexpr double tol = 1e-10;
+
+  auto nrConfig =
+      Ikarus::NewtonRaphsonConfig<decltype(linSolver)>{.parameters = {.tol = tol}, .linearSolver = linSolver};
+  auto nonLinearSolverObserver = std::make_shared<NonLinearSolverLogger>();
+  auto pathFollowingObserver   = std::make_shared<ControlLogger>();
+  auto vtkWriter =
+      std::make_shared<ControlSubsamplingVertexVTKWriter<std::remove_cvref_t<decltype(basis.flat())>>>(basis.flat(), d);
+  vtkWriter->setFileNamePrefix("CantileverNonlinearEAS");
+  vtkWriter->setFieldInfo("Displacement", Dune::VTK::FieldInfo::Type::vector, 2);
+  auto nr = createNonlinearSolver(nrConfig, nonOp);
+  auto lc = LoadControl(nr, 20, {0, 1});
+  nr->subscribeAll(nonLinearSolverObserver);
+  lc.subscribeAll({pathFollowingObserver, vtkWriter});
+
+  const auto controlInfo = lc.run();
+
+  double expectedLambda  = 1.0;
+  double expectedMaxDisp = 4.459851990227056; // abs(maxDisp) in ANSYS APDL = 4.48777
```

Comment:
I am not sure if this minor difference is acceptable here as ANSYS APDL is used more like a black box, where the control routine and nonlinear solver algorithms are not fully clear. Since the `easAutoDiffTest` passes, I think the solution should be correct here.